### PR TITLE
Fix doxygen errors in v3 MQTT documentation

### PR DESCRIPTION
**Pull request by gordonwang0**
Resolves doxygen issues in v3 MQTT documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws_iot_mqtt_client.h [hunk 1]
```diff
@@ -61,6 +61,7 @@ extern "C" {
 #include "threads_interface.h"
 #endif
 
+/** Greatest packet identifier, per MQTT spec */
```
Comment (sarenameas) [inline include/aws_iot_mqtt_client.h:64]: Do we need periods here?

Comment (gordonwang0) [reply include/aws_iot_mqtt_client.h:64]: The documentation for this version does not put periods at the end of lines.  So for consistency, no.

Comment (sarenameas) [reply include/aws_iot_mqtt_client.h:64]: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/include/types/iot_mqtt_types.h#L1129

Comment (gordonwang0) [reply include/aws_iot_mqtt_client.h:64]: That's v4 though.  v3 does not do that.

**Pull request merged**